src/app/core/component.ts

```diff
@@ -1,16 +1,31 @@
 import Params from "./params";
 import CustomComponent from "./customComponent";
 import ComponentPort from "./componentPort";
+import Feature from './feature'
 
 import Registry from "./registry";
 import * as FeatureRenderer2D from "../view/render2D/featureRenderer2D";
 import Port from "../library/port";
+import { ComponentPortInterchangeV1, ComponentInterchangeV1 } from "./init";
+import { ConnectionInterchangeV1, Point } from "./init";
 
 /**
  * This class contains the component abstraction used in the interchange format and the
  * high level device model of the microfluidic.
  */
 export default class Component {
+    protected _params: Params;
+    protected _name: string;
+    protected _id: string;
+    protected _type: string;
+    protected _entity: string;
+    protected _features: Feature[]; // Not sure if it's Feature[] or string[]
```

Comment:
We should figure this out soon easiest way to check this is to see where we are accessing this and then check what types are getting passed